crates/scarb-api/src/lib.rs

```diff
@@ -75,64 +80,150 @@ fn artifacts_for_package(path: &Utf8Path) -> Result<StarknetArtifacts> {
     Ok(starknet_artifacts)
 }
 
-/// Try getting the path to `starknet_artifacts.json` file that is generated by `scarb build` or `scarb build --test` commands.
-/// If contract artifacts are produced as part of the test target and exist in both `unittest` and `integrationtest`, then the path to `integrationtest` will be returned.
-/// If the file is not present, `None` is returned.
-fn get_starknet_artifacts_path(
-    target_dir: &Utf8Path,
-    target_name: &str,
-    current_profile: &str,
-    use_test_target_contracts: bool,
-) -> Option<Utf8PathBuf> {
-    let starknet_artifacts_file_name = |test_type: Option<&str>| -> String {
-        if let Some(test_type) = test_type {
-            format!("{target_name}_{test_type}test.test.starknet_artifacts.json")
-        } else {
-            format!("{target_name}.starknet_artifacts.json")
+#[derive(PartialEq, Debug)]
+struct StarknetArtifactsFiles {
+    base_file: Utf8PathBuf,
+    other_files: Vec<Utf8PathBuf>,
+}
+
+impl StarknetArtifactsFiles {
+    fn load_contracts_artifacts(
+        self,
+    ) -> Result<HashMap<String, (StarknetContractArtifacts, Utf8PathBuf)>> {
+        let mut base_artifacts = load_contracts_artifacts_and_source_sierra_paths(&self.base_file)?;
+
+        let compiled_artifacts = self
+            .other_files
+            .par_iter()
+            .map(load_contracts_artifacts_and_source_sierra_paths)
+            .collect::<Result<Vec<_>>>()?;
+
+        for artifact in compiled_artifacts {
+            for (key, value) in artifact {
+                base_artifacts.entry(key).or_insert(value);
+            }
```

Comment:
It still compiles those files, just to most probably discard them anyway? 👀 
Can't we check if we need specific contract artifact without compiling it first?